### DISCUSSION TITLE
Import types in schema.js

### DISCUSSION THREAD

**Discussion by perroudsky** [prisma/prisma#5784] [Q&A] [answered]
How can you import enums and models from the schema.prisma file to be used for the typeDefs in schema.js?

For exemple, if you add 
`enum PostType {
      MEDIA 
      PRODUCT
  }`
 to the schema.prisma how can you use it in the typedefs. 
If I try to do so I've like `Error: Unknown type "PostType". Did you mean "Post" or "__Type"?`

**Comment by ryands17** [comment, marked as answer]
@perroudsky 👋 

Enums are accessible by simply importing them. Given a schema like this:

```prisma
enum UserType {
  BASIC
  ADMIN
}

model User {
  id   Int      @id @default(autoincrement())
  name String?
  type UserType @default(BASIC)
}
```

I can import the enum constant as follows:

```ts
import { UserType } from '@prisma/client'

const BASIC = UserType.BASIC
```

Note here that the enum is not a type but a string literal. So you would need to use it in the following manner.

**Reply by perroudsky** [reply]
Hi @ryands17 ! thanks for the quick answer !

Got it but need more precisions :) 
How would you use it in this portion :

`
const typeDefs = `
type User {
 email: String!
 role: UserType
...
}

input UserCreateInput {
  email: String!
  role: UserType
...
}`
`

I know I've probably missed something but of course importing UserType still returns an error (undefined). 
In prisma 1 you could import all generate types with something like
`# import * from './generated/prisma.graphql'`

**Reply by ryands17** [reply]
These types cannot be used in GraphQL. GraphQL has its own type system and unlike Prisma 1, Prisma 2 is just an ORM and will give you TypeScript types. In GraphQL `typedefs` you would need to create an enum explicitly and use that.

**Reply by perroudsky** [reply]
So what's the prisma way to do that ? No more typeDefs and use nexus instead ?

**Reply by ryands17** [reply]
You can have a look at [this](https://github.com/prisma/prisma-examples/tree/latest/javascript/graphql-sdl-first) example from our examples repo of how to use Prisma with GraphQL.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
